v3-certificate/generate-certificate

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+#set -e
+
+# Checking openssl
+if [[ -z $(which openssl) ]]; then
+  echo "Please download openssl and make sure it is available in your PATH variable"
+  echo "Homepage: https://www.openssl.org/"
+  exit 1
+fi
+
+# Checking jq
+if [[ -z $(which jq) ]]; then
+  echo "Please download jq and make sure it is available in your PATH variable"
+  echo "jq is used to parse JSON elements"
+  echo "Homepage: https://stedolan.github.io/jq/"
+  exit 1
+fi
+
+# FusionAuth URL
+read -r -p "Your FusionAuth instance URL (with scheme): " FA_URL
+regex='^https?:\/\/[^\/]+(:[0-9]+\/)?$'
+if ! [[ $FA_URL =~ $regex ]]; then
+  echo "FusionAuth URL is not valid. Please make sure you have included the scheme (http:// or https://)."
+  exit 2
+fi
+
+# Extracting domain from URL
+FA_DOMAIN=$(echo "$FA_URL" | awk -F/ '{print $3}')
+if [[ -z "$FA_DOMAIN" ]]; then
+  echo "FusionAuth URL is not valid. Please make sure you have included the scheme (http:// or https://)."
+  exit 2
+fi
+
+# FusionAuth API Key
+read -r -p "API Key with /api/key/import endpoint:      " API_KEY
+API_KEY_LENGTH=${#API_KEY}
+if [ "$API_KEY_LENGTH" -ne 56 ]; then
+  echo "Your API Key seems invalid. Please make sure you copied it correctly."
+  exit 2
```

Comment:
Why `exit 2`?

Reply:
I always exit 1, but maybe I'm doing it wrong.

Reply:
You can exit anything other than 0 to indicate error. It is a good practice to group the errors to indicate similar erros. In this script, 1 means unmet requirements, 2 means bad input, etc...

Reply:
okay, no worries, seems fine to me.